cmd/prepare-engine.go

```diff
@@ -63,21 +65,30 @@ var prepareEngineWindowsCmd = &cobra.Command{
 }
 
 func initPrepareEngineParameters(targetOS string) {
+	validatePrepareEngineParameters(targetOS)
 	if prepareDebugMode {
 		prepareBuildModes = append(prepareBuildModes, build.DebugMode)
 	}
 	if prepareReleaseMode {
 		prepareBuildModes = append(prepareBuildModes, build.ReleaseMode)
 	}
-	validatePrepareEngineParameters(targetOS)
+	if prepareProfileMode {
+		prepareBuildModes = append(prepareBuildModes, build.ProfileMode)
+	}
 }
 
 func validatePrepareEngineParameters(targetOS string) {
+	if !prepareDebugMode && !prepareReleaseMode && !prepareProfileMode {
+		log.Errorf("No target mode set. Please select exactly one from: debug, profile, release.")
+		os.Exit(1)
+	}
+	if (prepareDebugMode && prepareReleaseMode) || (prepareDebugMode && prepareProfileMode) || (prepareReleaseMode && prepareProfileMode) {
```

Comment:
I think it still breaks when specifying all three flags. Maybe take a look at https://github.com/go-flutter-desktop/hover/blob/master/cmd/build.go#L311-L323 or maybe there is even a better way for that

Reply:
This line would not break under that circumstance, but I can switch to that way if it is preferred.

`(true && true) || (true & true) || (true && true)) == true`

Reply:
Ah yes you are right. Sorry didn't think about it properly

Reply:
Well now that you changed it anyway, we can leave it, although your previous implementation makes more sense :woman_shrugging: You can also drop the comment if you want to.